mne/surface.py

```diff
@@ -1697,6 +1699,194 @@ def marching_cubes(image, level):
+                         'is incorrect or improperly formatted, '
+                         f'got {op.join(subjects_dir, subject)}')
+
+
+def _check_image(image, subject, subjects_dir):
+    """Check if the image is in Freesurfer surface RAS space."""
+    import nibabel as nib
+    if not isinstance(image, nib.spatialimages.SpatialImage):
+        image = nib.load(image)
+    image = nib.MGHImage(
+        _get_img_fdata(image).astype(np.float32), image.affine)
+    fs_t1 = nib.load(op.join(subjects_dir, subject, 'mri', 'T1.mgz'))
+    if not np.allclose(image.affine, fs_t1.affine, atol=1e-6):
+        from nib.processing import resample_from_to
+        image = resample_from_to(image, (fs_t1.shape, fs_t1.affine))
+    return image, fs_t1
+
+
+def _check_dig(dig):
+    """Check if dig is in the coordinate frame."""
+    use_idx = [i for i, d in enumerate(dig)
+               if d['kind'] == FIFF.FIFFV_POINT_EEG]
+    dig = [dig[idx] for idx in use_idx]
+    if not dig:
+        raise ValueError('No electrophysiolgy channels found. '
+                         'Accepted types are "eeg", "ecog", "seeg" '
+                         'and "dbs". Make sure the channel types are '
+                         'set properly, see `mne.io.Raw.set_channel_types`')
+    ch_coords = np.array([ch['r'] for ch in dig])
+    coord_frame = dig[0]['coord_frame']
+    for d in dig[1:]:
+        if d['coord_frame'] != coord_frame:
+            raise ValueError(
+                'Inconsistent dig montage coordinate frame in '
+                f'``info``, got {coord_frame} and ' + str(d['coord_frame']))
+    if coord_frame != FIFF.FIFFV_COORD_MRI:
+        raise RuntimeError('Coordinate frame not supported, expected '
+                           f'"mri", got {coord_frame}')
+    return ch_coords, use_idx
+
+
+def _ch_coords_to_vox(ch_coords, fs_t1):
+    """Convert channel coordinates from surface RAS to voxel space."""
+    ch_coords = apply_trans(
+        np.linalg.inv(fs_t1.header.get_vox2ras_tkr()), ch_coords * 1000)
+    return ch_coords
+
+
+def _ch_coords_to_surface_RAS(ch_coords, fs_t1):
+    """Convert back to the surface RAS coordinate frame."""
+    ch_coords = apply_trans(
+        fs_t1.header.get_vox2ras_tkr(), ch_coords) / 1000
+    return ch_coords
+
+
+def _warn_missing_chs(montage, dig_image, after_warp):
+    """Warn that channels are missing."""
+    # ensure that each electrode contact was marked in at least one voxel
+    missing = set(np.arange(1, len(montage.ch_names) + 1)).difference(
+        set(np.unique(_get_img_fdata(dig_image))))
+    missing_ch = [montage.ch_names[idx - 1] for idx in missing]
+    if missing_ch:
+        warn('Channels ' + ', '.join(missing_ch) + ' were not assigned '
+             'voxels' + (' after applying SDR warp' if after_warp else ''))
+
+
+@fill_doc
+def warp_montage_volume(montage, image, reg_affine, sdr_morph,
+                        subject, template, subjects_dir=None,
```

Comment:
let's default to `temalpte='fsaverage'`. It's a reasonable choice and I assume many people will want it